lib/assets/traits/device_item_icon.dart

```diff
@@ -12,13 +12,22 @@ class DeviceItemIcon {
     Trait determiningTrait = traits[0];
     switch (determiningTrait.runtimeType) {
       case LockTrait:
-        return LockIcon(determiningTrait.state.value);
+        return LockIcon(determiningTrait.states
```

Comment:
What if the State is not available? `firstWhere` tries to find the first element that satisfies the predicate, but what if the State is not available (e.g. Property `supportsIsJammed` is `false`, if the `IsJammedState` is not available, would it fail here?

Reply:
Yeah, this would fail in this case.

Reply:
I think the assumption was taken on that a lock will know if it's locked, a thermostat has a target temperature (this is a lot less certain!), power traits know if they are on, and a battery level would know it's battery level as they are the core states used to display them. I guess the question is should we set a sensible default on each one? firstWhere has a orElse. Without the orElse it will throw an exception if nothing is found.